### PR TITLE
Port to Dashing: melodic-devel 2.5.1 to 2.6.2

### PR DESCRIPTION
**Pull request by mabelzhang**
In sequence of #521 and #523, this PR ports commits in the diffs of https://github.com/cra-ros-pkg/robot_localization/compare/dashing-devel...melodic-devel.

Specifically, this PR ports commits in the melodic [CHANGELOG](https://github.com/cra-ros-pkg/robot_localization/blob/melodic-devel/CHANGELOG.rst) items of 2.5.1 to 2.6.2, inclusive. Rough equivalent of 
```
git cherry-pick 6dbbf3e^..82a1ad1
```
but skipping the `CHANGELOG` and version bump commits.

I'm grouping these together because the next commit in sequence, https://github.com/cra-ros-pkg/robot_localization/commit/58bef056cf84856eb5e91a89f643017c448669ff, changes actual math, and I don't want to mess Dashing users up in case that changes behavior. I'll put that in a separate PR later.

We are actively testing `dashing-devel` branch on a robot as things are merged. If there are problems in the porting, hopefully we spot them.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,10 +5,6 @@ if(NOT WIN32)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
 endif()
 
-if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```
Comment (SteveMacenski) [inline CMakeLists.txt]: I think we want this?

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
@mabelzhang awesome -- I'm curious, who is "we"? Is this in the context of a mobile robot navigation task (using Navigation2?)

Literally the oldest ticket in our book is testing Nav2 with RL https://github.com/ros-planning/navigation2/issues/40 so if you're up to that, it would be good to know to close out

File: CMakeLists.txt (hunk 1 above)
Comment (mabelzhang) [reply CMakeLists.txt]: Yes, error corrected in 7f9e1b3

**Comment by IanTheEngineer** [comment]
@SteveMacenski, I'm working with @mabelzhang for the sync between melodic-ros2 branches but Mabel has been doing the actual port.

> Is this in the context of a mobile robot navigation task ?

Yes. I'll likely be able to provide details in a few months time.

**Comment by SteveMacenski** [comment]
> ToyotaResearchInstitute

Ah, I think I know already from the TSC meetings, got it. In that case, will close the Nav2 ticket.

Anyhow, I think this is good. I'm just letting it sit for a day in case someone has thoughts to add. Otherwise, will merge tomorrow.

**Comment by SteveMacenski** [comment]
@mabelzhang I saw you push to this recently. Is this good to go now? It's soaked long enough I'll merge if done.

**Comment by mabelzhang** [comment]
Yep, good to go.
The test fix is a fix in how they are run in CMakeLists.txt. Should have gone in earlier if I had noticed in #521.

**Pull request merged**